src/built_ins.md

```diff
@@ -18,222 +18,757 @@ The built-in types for `alan` are:
 * `bool`
 * `string`
 * `function`
-* `operator`
 * `Array<V>`
-* `Map<K, V>`
+* `HashMap<K, V>`
 * `KeyVal<K, V>`
-* `Tree<V>`
-
-(At the time of writing, the Tree type is not functional)
-
-There are no unsigned integers at this time due to the lack of such a primitive in Java. They will be added at some point in the future.
+* `Error`
+* `Maybe<T>`
+* `Result<T>`
+* `Either<T, U>`
 
 The `int64` and `float64` types are special among the numeric types, as these are the types that any numeric constant will be represented as, depending on whether or not it has a decimal.
 
-#### Built-in Variable
-
-There is *one* special built-in variable in the language, `_`. It is of type `void` and is used to throw away a value you don't want to save. This is only necessary if you have an operator-containing statement you wish to run but not pass to a function or anything else. This is a very esoteric need, since you could use the function form of one of the operators, but this can be desirable in certain situations.
-
 #### Built-in Interfaces
 
 There are a few built-in interfaces meant for working with several built-in types and functions.
 
-* `any` (an empty interface, it can match any value, but you can only accept it and pass it along to something else. Useful for logical "glue" functions like pair, cond, map, reduce, etc.)
-* `Array<any>` (an array of some kind of value)
-* `Stringifiable` (an interface with the single function definition of `fn toString(Stringifiable): string` to indicate that this type can be converted to a string)
+```rust
+// An empty interface, it can match any value, but you can only accept it and pass it along to
+// something else. Useful for logical "glue" functions like `pair`, `cond`, `map`, `reduce`, etc.
+interface any {}
+```
+
+```rust
+// A second empty interface, useful to allow functions to declare that it accepts two arguments of
+// any kind and they don't need to match each other.
+interface anythingElse {}
+```
```

Comment:
nit: let's include a short example for these or a link to `interfaces.md`? it is not abundantly clear exactly how it is used if you have never seen `alan`

Reply:
Sure, I think linking the `Interfaces` section is a good idea for not just this but for all of the sections. (Cross-referencing is generally good.)